src/app/shared/components/conceptos/conceptos.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, Renderer2, ViewChildren } from '@angular/core';
+import {
+  NbGetters,
+  NbSortDirection,
+  NbTreeGridRowComponent,
+  NbTreeGridDataSource,
+  NbTreeGridDataSourceBuilder,
+  NbSortRequest,
+} from '@nebular/theme';
+import { Store } from '@ngrx/store';
+import { LoadNodoSeleccionadoConcepto } from '../../actions/shared.actions';
+import { EstructuraArbolRubrosApropiaciones } from '../../interfaces/interfaces';
+import { seleccionarConceptos } from '../../selectors/shared.selectors';
+import { FORM_NODO_CONCEPTO } from './form_nodo_concepto';
+
+@Component({
+  selector: 'ngx-conceptos',
+  templateUrl: './conceptos.component.html',
+  styleUrls: ['./conceptos.component.scss']
+})
+export class ConceptosComponent implements OnInit {
+
+  @Output() ConceptoSeleccionado = new EventEmitter();
+  customColumn = 'Codigo';
+  defaultColumns = ['Nombre'];
+  allColumns = [this.customColumn, ...this.defaultColumns];
+  sortColumn: string;
+  sortDirection: NbSortDirection = NbSortDirection.NONE;
+  formData: any;
+  nodeData: any;
+  conceptos: any;
+  subConceptos$: any;
+  selectedNodeData: any;
+  idHighlight: any;
+  oldHighlight: ElementRef;
+  dataSource: NbTreeGridDataSource<EstructuraArbolRubrosApropiaciones>;
+
+  constructor(
+    private renderer: Renderer2,
+    private store: Store<any>,
+    private dataSourceBuilder: NbTreeGridDataSourceBuilder<EstructuraArbolRubrosApropiaciones>,
+  ) { }
+
+  ngOnInit() {
+    this.loadTree();
+
+    this.formData = FORM_NODO_CONCEPTO;
+    this.nodeData = undefined;
+  }
+
+  async onSelect(selectedItem: any, treegrid) {
+    this.idHighlight = treegrid.elementRef.nativeElement.getAttribute('data-picker');
+    this.ConceptoSeleccionado.emit(selectedItem.data);
+    if (selectedItem.children == null) this.store.dispatch(LoadNodoSeleccionadoConcepto(selectedItem.data));
+    this.selectedNodeData = selectedItem.data;
+  }
+
+  loadTree() {
+    const getters: NbGetters<any, any> = {
+      dataGetter: (node: any) => node.data || undefined,
+      childrenGetter: (node: any) => !!node.children && !!node.children.length ? node.children : [],
```

Comment:
Esta línea, en teoría se podría dejar simplemente como
```suggestion
      childrenGetter: (node: any) => node.children && node.children.length ? node.children : [],
```